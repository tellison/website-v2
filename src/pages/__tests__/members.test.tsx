import React from 'react';
import { render, waitFor } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest'
import { axe } from 'vitest-axe';
import Members, { Head } from '../members';

vi.mock('../../util/shuffle', () => {
  return {
    shuffle: (array) => {
      array = [
        {
          name: 'mock_member',
          logo: 'mock_logo.png',
          url: 'https://mock.com',
          tier: 'mock_tier',
        }
      ]
      return array
    }
  };
});

describe('Members page', () => {
  it('renders correctly', () => {
    const { container } = render(<Members />);

    // eslint-disable-next-line
    const pageContent = container.querySelector('main');

    expect(pageContent).toMatchSnapshot();
  });

  it('head renders correctly', () => {
    const { container } = render(<Head />);
    // eslint-disable-next-line
    const title = container.querySelector('title');
    expect(title?.textContent).toEqual('Adoptium Working Group Members | Adoptium');
  });

  it('has no accessibility violations', async () => {
    const { container } = render(<Members />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
