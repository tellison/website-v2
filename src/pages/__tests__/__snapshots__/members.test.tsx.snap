// Vitest Snapshot v1

exports[`Members page > renders correctly 1`] = `
<main>
  <section
    class="py-5 text-center container"
  >
    <div
      class="row py-lg-5"
    >
      <div
        class="col-lg-8 col-md-8 mx-auto"
      >
        <h1
          class="fw-light"
        >
          Adoptium® Working Group Members
        </h1>
        <h2
          class="fw-light pt-5"
        >
          Strategic Members
        </h2>
        <p
          class="lead text-muted"
        >
          Strategic Members are organizations that view Adoptium working group managed technology as critical to their organization’s future, and are investing significant resources to sustain and define the core activities that are the responsibility of the working group.
        </p>
        <div
          class="container"
        >
          <div
            class="row justify-content-center align-items-center"
          >
            <div
              class="col-lg-4 col-md-12 mt-5"
            >
              <a
                href="https://mock.com"
                rel="noopener noreferrer"
                target="_blank"
              >
                <img
                  alt="mock_member logo"
                  src="/images/mock_logo.png"
                  style="width: 10rem;"
                />
              </a>
            </div>
          </div>
        </div>
        <h2
          class="fw-light pt-5"
        >
          Enterprise Members
        </h2>
        <p
          class="lead text-muted"
        >
          Enterprise Members are typically organizations that view the Adoptium working group managed technology as a critical part of their organization‘s business operations. These organizations want to influence the direction and support the development of a runtime technology ecosystem through Eclipse Adoptium.
        </p>
        <div
          class="container"
        >
          <div
            class="row justify-content-center align-items-center"
          >
            <div
              class="col-lg-4 col-md-12 mt-5"
            >
              <a
                href="https://mock.com"
                rel="noopener noreferrer"
                target="_blank"
              >
                <img
                  alt="mock_member logo"
                  src="/images/mock_logo.png"
                  style="width: 10rem;"
                />
              </a>
            </div>
          </div>
        </div>
        <h2
          class="fw-light pt-5"
        >
          Participant Members
        </h2>
        <p
          class="lead text-muted"
        >
          Participant Members are typically organizations that deliver products or services based on Adoptium technology. These organizations want to participate in the evolution of the Eclipse Adoptium ecosystem to ensure it continues to meet their needs.
        </p>
        <div
          class="container"
        >
          <div
            class="row justify-content-center align-items-center"
          >
            <div
              class="col-lg-4 col-md-12 mt-5"
            >
              <a
                href="https://mock.com"
                rel="noopener noreferrer"
                target="_blank"
              >
                <img
                  alt="mock_member logo"
                  src="/images/mock_logo.png"
                  style="width: 10rem;"
                />
              </a>
            </div>
          </div>
        </div>
        <a
          class="btn btn-lg btn-primary mt-5"
          href="/join"
        >
          Want to Join?
        </a>
      </div>
    </div>
  </section>
</main>
`;
